File: leetcode/3111-minimum-rectangles-to-cover-points/minRectanglesToCoverPoints.ts
function minRectanglesToCoverPoints(points: number[][], w: number): number {
  points.sort((a, b) => a[0] - b[0]);
  let count = 0;
  let next = -1;

  for (const [x] of points) {
    if (x > next) {
      next = x + w;
      count++;
    }
  }

  return count;
}

export default minRectanglesToCoverPoints;

File: leetcode/3111-minimum-rectangles-to-cover-points/minRectanglesToCoverPoints.go
package leetcode

import "sort"

func minRectanglesToCoverPoints(points [][]int, w int) int {
	sort.Slice(points, func(i, j int) bool {
		return points[i][0] < points[j][0]
	})

	var count = 0
	var next = -1

	for _, point := range points {
		var x = point[0]
		if x > next {
			next = x + w
			count++
		}
	}
	return count
}
